app/controllers/application_controller.rb

```diff
@@ -2,9 +2,115 @@
 
 class ApplicationController < Sinatra::Base
 
-  configure do
-    set :public_folder, 'public'
-    set :views, 'app/views'
-  end
+ 	configure do
+	    set :public_folder, 'public'
+	    set :views, 'app/views'
+	    enable :sessions
+		set :session_secret, "password_security"
+	end
+
```

Comment:
This controller is overloaded. Consider separating your concerns by splitting some of these actions into separate controllers (i.e. a users controller and a tweets controller).